### DISCUSSION TITLE
Comment about current limitation on redeclare but not replaceable

### DISCUSSION THREAD

**Discussion by mpradovelasco** [OpenModelica/OpenModelica#11330] [Q&A] [answered]
Hi, 

After use the last version of OM 1.23 dev with the #11311 issue fixed, and although it seems to function properly, I want to point to a lack (in my opinion). Once a modifier with a redeclare has been selected trough the dialog box and the dialog window is closed, the following time that the dialog box will be opened the user cannot modify previous selection either consult which selection was done (excepting through the view tab). 

The only possibility is avoid the use of "annotation(choicesAllMatching = true)" and use choices where manually the one can keep the repleaceable clause. 

Thanks

**Comment by adeas31** [comment]
> Once a modifier with a redeclare has been selected trough the dialog box and the dialog window is closed, the following time that the dialog box will be opened the user cannot modify previous selection either consult which selection was done (excepting through the view tab).

Yeah, this is because now we check for `redeclare` since #11311. Maybe we should not check for it when it is defined in the component modification and only consider it when it is coming from component class or extends modification.

@perost @casella what do you think?

**Reply by mpradovelasco** [reply]
@adeas31 in my opinion, your suggestion is a fine solution.

**Reply by adeas31** [reply]
OK. #11343 implements that. Please check and let me know. If everything is fine then I will port it to 1.22

**Reply by casella** [reply]
LGTM. Please check with the next nightly and port it to 1.22

**Reply by mpradovelasco** [reply]
OK. I will check it with the next nightly version.

**Reply by mpradovelasco** [reply, marked as answer]
Hi again, 

I have just checked this issue successfully with the latest nightly version. 

Thanks!!
